staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/marshal.go

```diff
@@ -120,15 +120,16 @@ func (s *JSONSchemaPropsOrArray) UnmarshalJSON(data []byte) error {
 
 func (s JSON) MarshalJSON() ([]byte, error) {
 	if len(s.Raw) > 0 {
-		return json.Marshal(s.Raw)
+		return s.Raw, nil
 	}
 	return []byte("null"), nil
 
 }
 
 func (s *JSON) UnmarshalJSON(data []byte) error {
-	if err := json.Unmarshal(data, &(s.Raw)); err != nil {
-		return err
+	if len(data) > 0 && string(data[0]) != "null" {
```

Comment:
data[0] ? should be string(data), shoulnd't it?

Reply:
yes :+1: